File: src/#230/solution.go
package solution

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func kthSmallest(root *TreeNode, k int) int {
	ans := 0
	order := 0

	helper(root, &order, k, &ans)
	return ans
}

func helper(root *TreeNode, order *int, k int, ans *int) {
	if root == nil {
		return
	}

	helper(root.Left, order, k, ans)
	*order += 1
	if *order == k {
		*ans = root.Val
		return
	}
	helper(root.Right, order, k, ans)

	return

}

File: src/#230/solution.js
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val;
  this.left = left === undefined ? null : left;
  this.right = right === undefined ? null : right;
}

var kthSmallest = function (root, k) {
  let ans = 0;
  let order = 0;
  helper(root, k);

  return ans;

  function helper(root, k) {
    if (root === null) {
      return;
    }
    helper(root.left, k);
    order++;
    if (order === k) {
      ans = root.val;
      return;
    }
    helper(root.right, k);
  }
};
